catalog-rest-service/src/main/java/org/openmetadata/catalog/security/NoopAuthorizer.java

```diff
@@ -53,4 +73,38 @@ public boolean isAdmin(AuthenticationContext ctx) {
   public boolean isBot(AuthenticationContext ctx) {
     return true;
   }
+
+  private void addAnonymousUser() {
+    EntityUtil.Fields fields = new EntityUtil.Fields(FIELD_LIST, fieldsParam);
+    try {
+      User user = userRepository.getByName(null, username, fields);
+      if (user != null && (user.getIsBot() == null || !user.getIsBot())) {
+        user.setIsAdmin(true);
+      }
+      addOrUpdateUser(user);
+    } catch (EntityNotFoundException ex) {
+      User user =
+          new User()
+              .withId(UUID.randomUUID())
+              .withName(username)
+              .withEmail(username + "@domain.com")
```

Comment:
Wasn't the domain specified in the config file?
If so, should that be used?

Reply:
Yes, for anonymous user, we can simply use any default domain name.